addons/l10n_in/models/res_partner.py

```diff
@@ -28,6 +28,30 @@ class ResPartner(models.Model):
     )
 
     display_pan_warning = fields.Boolean(string="Display pan warning", compute="_compute_display_pan_warning")
+    l10n_in_gst_state_warning = fields.Char(compute="_compute_l10n_in_gst_state_warning")
+
+    @api.depends_context('company')
+    @api.depends('vat', 'state_id', 'country_id')
+    def _compute_l10n_in_gst_state_warning(self):
+        for partner in self:
+            if (
+                "IN" in partner.fiscal_country_codes
+                and partner.check_vat_in(partner.vat)
+            ):
+                if partner.vat[:2] == "99":
+                    partner.l10n_in_gst_state_warning = _(
+                        "As per GSTN the country should be other than India, so it's recommended to update it."
+                    )
+                else:
+                    state_id = self.env['res.country.state'].search([('l10n_in_tin', '=', partner.vat[:2])], limit=1)
```

Comment:
Not directly related to this PR, but shouldn't there be a unique constraint on `l10n_in_tin`? Then the limit is not necessary, and there is less confusion possible